src/fluree/server/config.clj

```diff
@@ -79,10 +79,20 @@
               [:server ::server-config]
               [:connection ::connection]
               [:consensus ::consensus]
-              [:http ::http]]}))
+              [:http ::http]]
+
+    ::sid-migration [:map
+                     [:ledgers {:doc "Collection of ledger aliases to migrate."} [:sequential :string]]
+                     [:force {:optional true :doc "If true, run the migration regardless of whether the ledger has already been migrated."}
+                      :boolean]]
+
+    ::sid-migration-config [:map {:closed true}
+                            [:server ::server-config]
+                            [:connection ::connection]
+                            [:sid-migration ::sid-migration]]}))
 
 (def coerce
-  (m/coercer ::config transform/string-transformer {:registry registry}))
+  (m/coercer [:or ::config ::sid-migration-config] transform/string-transformer {:registry registry}))
```

Comment:
I think these configs would be more extensible going forward as a [multi schema](https://github.com/metosin/malli?tab=readme-ov-file#multi-schemas) instead of `::or`. The initial dispatch function could just be this:
```clojure
(fn [scm]
  (if (contains? scm "sidMigration")
    ::sid-migration-config
    ::config))
```
We could modify that function (probably by replacing the `if` with a `cond` and adding more `contains?` checks) as we add more ways of running the system and their corresponding config types. 
```

Reply:
Functionality wise in this situation? Not much. I was more thinking about extensibility and clarifying intent, but I'm fine with it if you prefer it this way.